x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/agent_activity_flyout.tsx

```diff
@@ -0,0 +1,470 @@
+                  </h1>
+                </EuiTitle>
+              </EuiFlexItem>
+              <EuiFlexItem>
+                <EuiText color="subdued">
+                  <p>
+                    <FormattedMessage
+                      id="xpack.fleet.agentActivityFlyout.activityLogText"
+                      defaultMessage="Activity log of Elastic Agent operations will appear here."
+                    />
+                  </p>
+                </EuiText>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiPanel>
+        </EuiFlyoutHeader>
+
+        <EuiFlyoutBody>
+          {currentActionsEnriched.length === 0 ? (
+            <EuiPanel hasBorder={true} borderRadius="none">
+              <EuiFlexGroup
+                direction="column"
+                gutterSize="m"
+                justifyContent={'center'}
+                alignItems={'center'}
+              >
+                <EuiFlexItem>
+                  <EuiText color="subdued">
+                    <FormattedMessage
+                      id="xpack.fleet.agentActivityFlyout.noActivityText"
+                      defaultMessage="No recent agent activity."
+                    />
+                  </EuiText>
+                </EuiFlexItem>
+              </EuiFlexGroup>
+            </EuiPanel>
+          ) : null}
+          {inProgressActions.length > 0 ? (
+            <ActivitySection
+              title={
+                <FormattedMessage
+                  id="xpack.fleet.agentActivityFlyout.inProgressTitle"
+                  defaultMessage="In progress"
+                />
+              }
+              actions={inProgressActions}
+              abortUpgrade={abortUpgrade}
+            />
+          ) : null}
+          {todayActions.length > 0 ? (
+            <ActivitySection
+              title={
+                <FormattedMessage
+                  id="xpack.fleet.agentActivityFlyout.todayTitle"
+                  defaultMessage="Today"
+                />
+              }
+              actions={todayActions}
+              abortUpgrade={abortUpgrade}
+            />
+          ) : null}
+          {Object.keys(otherDays).map((day) => (
+            <ActivitySection
+              title={<FormattedDate value={day} year="numeric" month="short" day="2-digit" />}
+              actions={otherDays[day]}
+              abortUpgrade={abortUpgrade}
+            />
+          ))}
+        </EuiFlyoutBody>
+      </EuiFlyout>
+    </>
+  );
+};
+
+const ActivitySection: React.FunctionComponent<{
+  title: ReactNode;
+  actions: ActionStatus[];
+  abortUpgrade: (action: ActionStatus) => Promise<void>;
+}> = ({ title, actions, abortUpgrade }) => {
+  return (
+    <>
+      <EuiPanel color="subdued" hasBorder={true} borderRadius="none">
+        <EuiText>
+          <b>{title}</b>
+        </EuiText>
+      </EuiPanel>
+      {actions.map((currentAction) =>
+        currentAction.type === 'UPGRADE' && currentAction.status === 'in progress' ? (
```

Comment:
Higher level question, but is it intentional that some string values for things like `type` are `SCREAMING_SNAKE_CASE` but others are just `lower case strings with spaces`? Seems a bit inconsistent.

Reply:
you're right, I can make this more consistent